pysat/utils/_core.py

```diff
@@ -224,217 +224,152 @@ def load_netcdf4(fnames=None, strict_meta=False, file_format=None,
+                                               fill_label=fill_label)
+
+                    for key, clean_key in zip(obj_var_keys, clean_var_keys):
+                        # store attributes in metadata, exept for dim name
+                        meta_dict = {}
+                        for nc_key in data.variables[key].ncattrs():
+                            meta_dict[nc_key] = \
+                                data.variables[key].getncattr(nc_key)
+                        dim_meta_data[clean_key] = meta_dict
+
+                    dim_meta_dict = {'meta': dim_meta_data}
+                    if index_key_name is not None:
+                        # add top level meta
+                        for nc_key in data.variables[obj_key_name].ncattrs():
+                            dim_meta_dict[nc_key] = \
+                                data.variables[obj_key_name].getncattr(nc_key)
+                        mdata[obj_key_name] = dim_meta_dict
+
+                    # iterate over all variables with this dimension
                     # data storage, whole shebang
                     loop_dict = {}
                     # list holds a series of slices, parsed from dict above
                     loop_list = []
-                    loop_dict[obj_key_name] = \
-                        data.variables[obj_key_name][:, :, :]
+                    for key, clean_key in zip(obj_var_keys, clean_var_keys):
+                        # data
+                        loop_dict[clean_key] = \
+                            data.variables[key][:, :].flatten(order='C')
                     # number of values in time
-                    loop_lim = data.variables[obj_key_name].shape[0]
+                    loop_lim = data.variables[obj_var_keys[0]].shape[0]
                     # number of values per time
-                    step_size_x = len(data.variables[obj_key_name][0, :, 0])
-                    step_size_y = len(data.variables[obj_key_name][0, 0, :])
-                    step_size = step_size_x
-                    loop_dict[obj_key_name] = \
-                        loop_dict[obj_key_name].reshape((loop_lim*step_size_x,
-                                                         step_size_y))
+                    step_size = len(data.variables[obj_var_keys[0]][0, :])
                     # check if there is an index we should use
                     if not (index_key_name is None):
                         # an index was found
                         time_var = loop_dict.pop(index_key_name)
                         if time_index_flag:
                             # create datetime index from data
-                            if file_format == 'NETCDF4':
-                                time_var = pds.to_datetime(1E6*time_var)
-                            else:
-                                time_var = pds.to_datetime(1E6*time_var)
+                            time_var = pds.to_datetime(1E6 * time_var)
```

Comment:
There are several bits like this that I've simplified in the new code. Should bump our coverage up a bit.